tests/rados/test_config_parameter_chk.py

```diff
@@ -0,0 +1,142 @@
+"""
+from configparser import ConfigParser
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from tests.rados.monitor_configurations import MonConfigMethods
+from utility.log import Log
+
+log = Log(__name__)
+ini_file = "conf/quincy/rados/test-confs/rados_config_parameters.ini"
+
+
+def run(ceph_cluster, **kw):
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    osd_list = []
+    mclock_profile = ["balanced", "high_recovery_ops", "high_client_ops"]
+    # To get the parameters from the ini
+    rados_object = RadosOrchestrator(node=cephadm)
+    mon_object = MonConfigMethods(rados_obj=rados_object)
+
+    config_info = ConfigParser()
+    config_info.read(ini_file)
+
+    # To get the OSD list
+    ceph_nodes = kw.get("ceph_nodes")
+    for node in ceph_nodes:
+        if node.role == "osd":
+            node_osds = rados_object.collect_osd_daemon_ids(node)
+            osd_list = osd_list + node_osds
+    if config.get("scenario") == "msgrv2_5x":
+        msgr_parameter = config_info.items("MSGRV2")
+        result = config_param_checker(mon_object, osd_list, msgr_parameter)
+        if result:
+            log.error("The MSGRV2 parameters are exist in the 5.x build")
+            return 1
+        return 0
+    if config.get("scenario") == "msgrv2_6x":
+        msgr_parameter = config_info.items("MSGRV2")
+        result = config_param_checker(mon_object, osd_list, msgr_parameter)
+        if not result:
+            log.error("The MSGRV2 parameter value is not correct in the 6.x build")
+            return 1
+        return 0
+
+    # Checking all sleep parameters in mclock profiles
+    if config.get("scenario") == "mclock_sleep":
+        mclock_sleep_parameters = config_info.items("Mclock_sleep")
+        for profile in mclock_profile:
+            mon_object.set_config(
+                section="osd", name="osd_mclock_profile", value=profile
+            )
+            # setting the  sleep parameters from 0 to 0.1
+            for param in mclock_sleep_parameters:
+                mon_object.set_config(section="osd", name=param[0], value="0.100000")
+            # Check that sleep parameters are not modifiable and the value is 0
+            result = config_param_checker(mon_object, osd_list, mclock_sleep_parameters)
+            if not result:
+                log.error(
+                    "The Mclock sleep parameter value is not correct in the 6.x build"
+                )
+                return 1
+        return 0
+
+    # Verifying the reservation, weight and limit parameters are not modifiable
+    # Currently a BZ raised on this feature. The number is BZ#2124137.
+    if config.get("scenario") == "mclock_chg_chk":
+        mclock_chg_parameters = config_info.items("Mclock_paramert_set")
+        mclock_default_parmeters = config_info.items("Mclock_default")
```

Comment:
Same as above. Add  :
Try without override, the Values should be modifiable.
Try with override, the Values should be modifiable.

Post this check, revert back to original values.

Reply:
Updated the code as per the review comments.